src/file/imp/unix.rs

```diff
@@ -94,11 +104,20 @@ pub fn reopen(file: &File, path: &Path) -> io::Result<File> {
     let new_file = OpenOptions::new().read(true).write(true).open(path)?;
     let old_meta = file.metadata()?;
     let new_meta = new_file.metadata()?;
-    if old_meta.dev() != new_meta.dev() || old_meta.ino() != new_meta.ino() {
-        return Err(io::Error::new(
-            io::ErrorKind::NotFound,
-            "original tempfile has been replaced",
-        ));
+    cfg_if! {
+        if #[cfg(any(not(target_os = "wasi"), feature = "nightly"))] {
```

Comment:
Let's do this at the function level. That is, define one function that always returns an error, and one function that actually works.

Reply:
Also, does this actually _work_ on wasi? I'd expect dev/ino to return garbage.

Reply:
Yeah it does, dev/ino are parts of WASI API and are propagated from WASI runtime. https://github.com/WebAssembly/WASI/blob/main/phases/snapshot/docs.md#-filestat-struct